### PR TITLE
Reintroduce tests

### PR DESCRIPTION
**Pull request by maksymiuks**

### CODE REVIEW

File: R/check_process.R [hunk 1]
```diff
@@ -1,17 +1,35 @@
...
+  # Case 2: status token is on the next line:
+  # consume remainder of current line + newline + optional indent,
+  # but only if the next thing is a status token at end-of-line
+  "[^\\n]*\\n[ \\t]*(?=(?:[A-Z]{2}[A-Z0-9_-]*)\\s*(?:\\n|$))",
+  "|",
+  # Case 3: no status token (eat remainder/comment and stop here)
+  "[^\\n]*",
+  ")",
+  "(?<status>(?:[A-Z]{2}[A-Z0-9_-]*)|)", # capture status token, or capture empty string if absent
+  "(?=\\s*(?:\\n|$))"                    # must end at newline/end (allow trailing whitespace)
```
Comment (maksymiuks) [inline R/check_process.R:30]: The previous regexp did not match the status of the test in certain edge cases. Especially where some additional content (like a description of which test file is run). The other problem was that the output passed to this regexp is a stream, and depending on when that stream was consumed, the output could not have the expected form. Therefore, now we have that  monstrosity, which however should be roboutes to any variations on that stream

File: R/check_process.R [hunk 2]
```diff
@@ -46,9 +64,17 @@ check_process <- R6::R6Class(
...
+      # to call it explicitly
+      checks <- self$checks
+      # In some cases, check subprocess might suffer from a race condition, when
+      # process itself finished, but the final results of the last subcheck
+      # are not yet available to parse. Therefore we allow the process to
+      # finalize only if the last subcheck has reported status.
+      if (checks[length(checks)] != "") {
+        self$save_results()
+        if (is.function(f <- private$finish_callback)) f(self)
+      }
```
Comment (maksymiuks) [inline R/check_process.R:83]: As the comments say, we had a race condition in our reported, and in some cases, the finisher (and thus parsing results to the file) was called before the process object consumed the output from the subprocess stream. In such cases, the serialized rcmdcheck file did not include test results (or more broadly, the last subcheck in the rcmd check). That's why we now make sure that the last subcheck was properly consumed by the process object, before we serialize the entire output. That also spawned some problems with the regexp, described above, because in fact the subchecks parsing mechanism was, in some cases, mismatching, but they are also fixed in this PR.

**Comment by maksymiuks** [comment]
@dgkf FYI! :)
